xbmc/pvr/PVRDatabase.cpp

```diff
@@ -661,8 +644,23 @@ bool CPVRDatabase::UpdateLastWatched(const CPVRChannel &channel)
 
 bool CPVRDatabase::UpdateLastWatched(const CPVRChannelGroup &group)
 {
-  std::string strQuery = PrepareSQL("UPDATE channelgroups SET iLastWatched = %d WHERE idGroup = %d",
+  const std::string strQuery = PrepareSQL("UPDATE channelgroups SET iLastWatched = %d WHERE idGroup = %d",
     group.LastWatched(), group.GroupID());
 
   return ExecuteQuery(strQuery);
 }
+
+int CPVRDatabase::GetChannelId(int iClientId, int iUniqueId)
+{
+  int channelId = -1;
+
+  const std::string strQuery = PrepareSQL("SELECT idChannel FROM channels WHERE iClientId = %d AND iUniqueId = %d",
+                                          iClientId, iUniqueId);
+
+  std::string value = GetSingleValue(strQuery);
```

Comment:
don't forget to call `m_pDS->close()` and please surround with try/catch like we did for other parts in `PVRDatabase`.